### PR TITLE
add union aggregator

### PR DESCRIPTION
**Pull request by mccanne**
This commit adds a union aggregator that takes union over all of its
input values and produces a value of type set of this computed union.
The type of the input values must be uniform.

### CODE REVIEW

File: reducer/reducer.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ var (
 	ErrFieldRequired = errors.New("field parameter required")
 )
 
-type Maker func() Interface
+type Maker func(*resolver.Context) Interface
```
Comment (henridf) [inline reducer/reducer.go:22]: This isn't a git conflict but runMany() in reducer_test.go also needs updating... else the build will fail after merging this.

Comment (mccanne) [reply reducer/reducer.go:22]: Fixed

**Review by henridf — COMMENTED** [review]
I think you mentioned a size limit / circuit breaker on the forthcoming collect reducer; doesn't this need one too? Or maybe in a later PR?

**Comment by mccanne** [comment]
> I think you mentioned a size limit / circuit breaker on the forthcoming collect reducer; doesn't this need one too? Or maybe in a later PR?

Yes, thanks!  I had the same thought and already added it to the upcoming branch.  I will bring it forward to this PR.

**Comment by mccanne** [comment]
I think this is ready now.

File: proc/groupby/row.go [hunk 2]
```diff
@@ -40,7 +40,8 @@ func (v valRow) ConsumePart(rec *zng.Record) error {
 		if !ok {
 			return errors.New("reducer row doesn't decompose")
 		}
-		v, err := col.nameExpr.Eval(rec)
+		resolver := col.nameExpr
+		v, err := resolver.Eval(rec)
```
Comment (nwt) [inline proc/groupby/row.go:44]:
```suggestion
		v, err := col.nameExpr.Eval(rec)
```

File: reducer/union.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		for u.size > MaxValueSize {
+			u.deleteOne()
+			u.MemExceeded++
+		}
+	}
+}
+
+func (u *Union) deleteOne() {
+	for key := range u.val {
+		u.size -= len([]byte(key))
```
Comment (nwt) [inline reducer/union.go:58]:
```suggestion
		u.size -= len(key)
```

File: reducer/union.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		delete(u.val, key)
+		return
+	}
+}
+
+func (u *Union) Result() zng.Value {
+	b := zcode.NewBuilder()
```
Comment (nwt) [inline reducer/union.go:65]:
```suggestion
	var b zcode.Builder
```

**Review by nwt — APPROVED** [review]

File: proc/groupby/row.go (hunk 2 above)
Comment (mccanne) [reply proc/groupby/row.go:44]: that looks familiar

File: reducer/union.go (hunk 3 above)
Comment (mccanne) [reply reducer/union.go:58]: Ok, I was thinking go did rune length on strings but I see it doesn't.  Thanks!

**Pull request merged**